DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Reports/ReportsController.cs

```diff
@@ -55,26 +61,92 @@ public IActionResult Index()
             return View(model);
         }
 
+        [NoCaching]
         [Route("Data")]
         public IEnumerable<ActivityDataRowModel> GetGraphData()
         {
             var centreId = User.GetCentreId();
             var adminId = User.GetAdminId()!.Value;
             var adminUser = userDataService.GetAdminUserById(adminId)!;
 
-            var filterData = Request.Cookies.ParseReportsFilterCookie(adminUser);
+            var filterData = Request.Cookies.RetrieveFilterDataFromCookie(adminUser);
 
             var activity = activityService.GetFilteredActivity(centreId, filterData!);
             return activity.Select(
                 p => new ActivityDataRowModel(p, DateHelper.GetFormatStringForGraphLabel(p.DateInformation.Interval))
             );
         }
 
-        // [HttpGet]
-        // [Route("EditFilters")]
-        // public IActionResult EditFilters()
-        // {
-        //     return View();
-        // }
+        [HttpGet]
+        [Route("EditFilters")]
+        public IActionResult EditFilters()
+        {
+            var centreId = User.GetCentreId();
+            var adminId = User.GetAdminId()!.Value;
+            var adminUser = userDataService.GetAdminUserById(adminId)!;
+
+            var filterData = Request.Cookies.RetrieveFilterDataFromCookie(adminUser);
+
+            var (jobGroups, courseCategories, courses) = GetDropdownValues(centreId, adminUser);
+
+            var dataStartDate = activityDataService.GetStartOfActivityForCentre(centreId);

```

Comment:
If we put this method in the activityService, we don't need to inject the data service in the controller